drivers/core_demographics_report/app/models/core_demographics_report/document_exports/core_demographics_excel_export.rb

```diff
@@ -15,57 +17,122 @@ def authorized?
+          'Race Overall' => [th_style, nil, nil, nil, percentage_format] + [nil, percentage_format] * coc_count,
+          'Ethnicity Overall' => [th_style, nil, nil, nil, percentage_format] + [nil, percentage_format] * coc_count,
+          'Relationship' => [th_style, nil, nil, nil, percentage_format],
+          'Disabilities' => [th_style, nil, nil, nil, percentage_format],
+          'At Least One Disability' => [th_style, nil, nil, nil, percentage_format],
+          'No Disability' => [th_style, nil, nil, nil, percentage_format],
+          'DV Response' => [th_style, nil, nil, nil, percentage_format],
+          'DV Occurrence Timing' => [th_style, nil, nil, nil, percentage_format],
+          'Number of Times Response' => [th_style, nil, nil, nil, percentage_format],
+          'Number of Months' => [th_style, nil, nil, nil, percentage_format],
+          'Prior Living Situation' => [th_style, nil, nil, nil, percentage_format],
+          'Household Types' => [th_style, nil, nil, nil, nil, percentage_format] + [nil, nil, percentage_format] * coc_count,
+          'Clients in Projects' => [th_style, nil, nil, nil, nil],
+        }
+        wb.add_worksheet(name: 'Data') do |sheet|
+          sheet.styles.add_style(sz: 24, b: true, alignment: { horizontal: :center })
+          report.class.data_for_export(reports).each_with_index do |(title, row), index|
+            if index.zero?
+              style = []
+            elsif index.positive?
+              if reports.count > 1
+                if title.starts_with?('_')
+                  title = ''
+                  style = [nil, report_style, nil, nil, nil, comparison_style, nil, nil, nil]
+                else
+                  style = [th_style, report_style, nil, nil, nil, comparison_style, nil, nil, nil]
+                end
+              else
+                style = [th_style, nil]
+              end
+              if title.starts_with?('*')
+                # Remove only the first character since we know it's an asterisk
+                title = title[1..]
+                style = Array.new(12 + (coc_count * 3), th_style)
+              elsif title.starts_with?('_')
+                if title.include?('_data_')
+                  key = title[0..title.index('_data_')].gsub('_', '')
+                  style = data_styles[key]
+                end
+                title = ''
+              end
+            end
 
-    private def write_tmp_file(data, file_name)
-      Dir.mktmpdir do |dir|
-        safe_name = file_name.gsub(/[^- a-z0-9]+/i, ' ').slice(0, 50).strip
-        file_path = "#{dir}/#{safe_name}.xlsx"
-        File.open(file_path, 'wb') { |file| file.write(data) }
-        yield(Pathname.new(file_path).open)
+            sheet.add_row(
+              [
+                title,
+                *row,
+              ],
+              style: style,
+            )
+          end
+        end
       end
-      true
     end
 
     protected def report_class
```

Comment:
I can't quite comment on the right line but looks like we can remove `controller_class` here (and in other controllers)